binance-pipeline/lambdas/binance_historical_extraction.py

```diff
@@ -0,0 +1,249 @@
+    trade_id = trade_id_finder.get_trade_id_for_date()
+
+    config = make_config(start_date, end_date)
+    orchestration_handler_generic(trade_id, symbol, config)
+
+    return
+
+
+def handle_from_start(symbol, end_date: str = ''):
+    trade_id = 0 # assumed true
+
+    config = make_config(end_date)
+    orchestration_handler_generic(trade_id, symbol, config)
+
+    return
+
+
+def orchestration_handler_generic(trade_id, symbol, config):
+    reverse = config['reverse']
+    starting_ids = get_starting_ids(trade_id, RESERVED_CONCURRENCY, CHUNK_LIMIT, reverse)
+    for id_ in starting_ids:
+        payload = create_payload(symbol, id_)
+        config['payload'] = payload
+
+        message = {"config": config, "type": "api_call"}
+        send_sqs_message(SQS_QUEUE_NAME_EXTRACTION, message)
+
+
+def make_candidate_id(trade_id: int, delta: int, reverse: bool=False):
+    if reverse:
+        candidate_id = trade_id - delta
+    else:
+        candidate_id = trade_id + delta
+
+    return candidate_id
+
+
+def get_starting_ids(trade_id, concurrency, chunk_size, reverse=False):
+    starting_ids = []
+    for i in range(1, concurrency + 1):
+        delta = i*chunk_size
+        if (candidate_id := make_candidate_id(trade_id, delta, reverse)) >= 0:
+            starting_ids.append(candidate_id)
+
+    return starting_ids
+
+
+def handle_api_call(config: dict):
+    start = time.perf_counter()
+    payload = config['payload']
+
+    binance_query_api = BinanceQueryAPI()
+
+    status_code, res = binance_query_api.old_trade_lookup(
+                payload['symbol'], CHUNK_LIMIT, payload['fromId']
+            )
+
+    if status_code in [418, 429]:
+        handle_rate_limit(status_code, res, config)
+
+    # TO DO: store result to file
+    #
+    #
+
+    else:
+        delta = RESERVED_CONCURRENCY * CHUNK_LIMIT
+        next_trade_id = make_candidate_id(payload['fromId'], delta, config['reverse'])
+
+        if (len(res) == CHUNK_LIMIT) and (next_trade_id >= 0):
+            config['payload']['fromId'] = next_trade_id
+            message = {'config': config, 'type': 'api_call'}
+            end = time.perf_counter()
+
+            runtime = end - start
+            delay_seconds = calculate_delay_seconds(runtime, RESERVED_CONCURRENCY, RATE_LIMIT)
+
+            if delay_seconds:
+                delay_config = get_delay_config(delay_seconds)
+                message = {
+                            "status_code": status_code,
+                            "delay_config": delay_config,
+                            "config": config
+                        }
+                send_rate_throttle_message(message)
+            else:
+                send_sqs_message(SQS_QUEUE_NAME_EXTRACTION, message)
+
+    return
+
+
+def calculate_delay_seconds(runtime: int, reserved_concurrency: int, rate_limit: int) -> int:
+    delay_seconds = math.ceil(((60 * reserved_concurrency)/rate_limit) - runtime)
+    delay_seconds = max(0, delay_seconds)
```

Comment:
one-liner

Reply:
for me the readability would suffer with one line - I am concerned that line 193 is hard to understand as it is

Reply:
Maybe just `return max(0, delay_seconds)` and chop up the complex bit to make it more understandable?